solanches/controller.py

```diff
@@ -1,15 +1,16 @@
 from . models import Produto, Comercio, Cardapio
 
 
-def cadastra_produto(nome, descricao, imagem, preco, categoria):
+def cadastra_produto(nome, descricao, imagem, preco, categoria, nome_comercio):
     assert nome and type(nome) is str, "Erro: nome inválido!"
     assert descricao and type(descricao) is str, "Erro: descricao inválida!"
     assert imagem and type(imagem) is str, "Erro: imagem inválida!"
     assert preco and type(preco) is float, "Erro: preco inválido!"
     assert categoria and type(categoria) is str, "Erro: categoria inválida!"
+    assert nome_comercio and type(nome_comercio) is str, "Erro: nome de comercio invalido"
 
     novo_produto = Produto(nome, descricao, preco, categoria, imagem)
-    novo_produto.save()
+    Comercio.add_produto(novo_produto, nome_comercio)
 
     return novo_produto.to_dict()
```

Comment:
Acho que é bom retornar o resultado do `Comercio.add_produto`, para garantir que o produto foi cadastrado e combinar com o retorno da camada `rest.py` que, pelo nome da variável tb esperava o id do produto como retorno.
```suggestion
    produto = Comercio.add_produto(novo_produto, nome_comercio)

    return produto
```